ush/forecast_postdet.sh

```diff
@@ -11,88 +11,72 @@
 ## for execution.
 #####
 
-DATM_postdet(){
-  ######################################################################
-  # Link DATM  inputs (ie forcing files)                           #
-  ######################################################################
-
-  #TODO: This should be some loop through CDATE-> CDATE+ FORECAST length
-  #and get input from either CFSR or GEFS or Whatever...
-  #Currently assumes you only need the month of DATM input for IC date
-  #DATMINPUTDIR should be machine specific
-
-  # DATM forcing file name convention is ${DATM_FILENAME_BASE}.$YYYYMMDDHH.nc
-  echo "Link DATM forcing files"
-  DATMINPUTDIR="/scratch2/NCEPDEV/marineda/DATM_INPUT/CFSR/${SYEAR}${SMONTH}"
-  $NLN -sf ${DATMINPUTDIR}/${DATM_FILENAME_BASE}*.nc $DATA/DATM_INPUT/
-}
-
-FV3_GFS_postdet(){
-  echo "SUB ${FUNCNAME[0]}: $RERUN and $warm_start determined for $RUN"
+FV3_postdet(){
+  echo "SUB ${FUNCNAME[0]}: Entering for RUN = ${RUN}"
 
-  echo $warm_start
-  echo $RERUN
+  echo "warm_start = ${warm_start}"
+  echo "RERUN = ${RERUN}"
 
   #-------------------------------------------------------
-  if [ $warm_start = ".true." -o $RERUN = "YES" ]; then
+  if [ "${warm_start}" = ".true." -o "${RERUN}" = "YES" ]; then
     #-------------------------------------------------------
     #.............................
-    if [ $RERUN = "NO" ]; then
+    if [[ ${RERUN} = "NO" ]]; then
       #.............................
 
       # Link all restart files from previous cycle
       for file in "${COM_ATMOS_RESTART_PREV}/${sPDY}.${scyc}0000."*.nc; do
-        file2=$(echo $(basename $file))
-        file2=$(echo $file2 | cut -d. -f3-) # remove the date from file
-        fsuf=$(echo $file2 | cut -d. -f1)
-        $NLN $file $DATA/INPUT/$file2
+        file2=$(echo $(basename "${file}"))
+        file2=$(echo "${file2}" | cut -d. -f3-) # remove the date from file
+        fsuf=$(echo "${file2}" | cut -d. -f1)
+        ${NLN} "${file}" "${DATA}/INPUT/${file2}"
       done
 
       # Replace sfc_data with sfcanl_data restart files from current cycle (if found)
-      if [ "${MODE}" = "cycled" ] && [ "${CCPP_SUITE}" = "FV3_GFS_v16" ]; then  # TODO: remove if statement when global_cycle can handle NOAHMP
+      if [[ "${MODE}" = "cycled" ]] && [[ "${CCPP_SUITE}" = "FV3_GFS_v16" ]]; then  # TODO: remove if statement when global_cycle can handle NOAHMP
         for file in "${COM_ATMOS_RESTART}/${sPDY}.${scyc}0000."*.nc; do
-          file2=$(echo $(basename $file))
-          file2=$(echo $file2 | cut -d. -f3-) # remove the date from file
-          fsufanl=$(echo $file2 | cut -d. -f1)
-          file2=$(echo $file2 | sed -e "s/sfcanl_data/sfc_data/g")
-          rm -f $DATA/INPUT/$file2
-          $NLN $file $DATA/INPUT/$file2
+          file2=$(echo $(basename "${file}"))
```

Comment:
The extra echo is redundant.
```suggestion
          file2=$(basename "${file}")
```